### ISSUE TITLE
#[warn(unused_assignments)] when variable only is compared

### ISSUE THREAD

**Issue by martinhath** [rust-lang/rust#21365]
By running the following code,
```rust
fn main(){
    let mut c = 0;
    let mut i = 0;
    loop {
        c = i;
        i += 1;
        if c == 10{
            break;
        }
    }
}
``` 
 I get this warning:
```
r.rs:2:9: 2:14 warning: value assigned to `c` is never read, #[warn(unused_assignments)] on by default
r.rs:2     let mut c = 0;
```

So, somehow comparing (and thus reading) a variable doesn't count as 'read'. 

```
rustc 1.0.0-nightly (3d0d9bb6f 2015-01-12 22:56:20 +0000)
binary: rustc
commit-hash: 3d0d9bb6fbdf6ead9396f0874cf38f3ef7c9be5c
commit-date: 2015-01-12 22:56:20 +0000
host: x86_64-unknown-linux-gnu
release: 1.0.0-nightly
```

**Comment by Gankro** [comment]
Litterally the value `0` that you assign to `c` is never read. I think it just wants you to do `let mut c;`. But even then you should just move the declaration of `c` inside the loop. `let c = i`. No mutability, no uninitialized values.

**Comment by steveklabnik** [comment]
Yes, this warning is correct.
